### PR TITLE
Add API Documentation links to previous versions table.

### PR DESCRIPTION
**Pull request by brycebaril**
I have had to url hack this so many times, was stoked that the io.js website had it, let's get it on nodejs.org as well.

selection_033

### CODE REVIEW

File: layouts/download-releases.hbs [hunk 1]
```diff
@@ -42,6 +42,9 @@
                                         <a href="{{changeloglink version}}">
                                             {{i18n 'links.pages.changelog'}}
                                         </a>
+                                        <a href="{{apidocslink version}}">
+                                            {{i18n 'releases.documentation'}}
```
Comment (Fishrock123) [inline layouts/download-releases.hbs]: I suggest using `docs.text` instead, for consistency. (defined here: https://github.com/nodejs/new.nodejs.org/pull/161/files#diff-7c141f233ad3e5f2b1dd471e292aec5dR59)

**Comment by brycebaril** [comment]
Updated with @Fishrock123's suggestion:

selection_034

**Comment by rvagg** [comment]
`+    return semver.satisfies(version, '>=1.0.0 <4.0.0') ?` not quite, 3.3.0+ does `docs` .. I'll manually add a symlink for 3.3.1 but a better solution might be to make a redirect for everything on iojs.org so this isn't necessary.

beside that, lgtm

**Comment by Fishrock123** [comment]
LGTM with Rod's nits.

**Comment by brycebaril** [comment]
Updated to use `docs` for all io.js releases as per @rvagg's new rewrite rules on the io.js website. Should be good to go

**Comment by fhemberger** [comment]
Cool, thanks!

**Pull request merged**